src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -76,6 +77,78 @@ protected virtual HttpClient GetHttpClient()
             return await ExecuteWithRetryAsync(uri, headers, body, HttpMethod.Post, logger, doNotThrow: true).ConfigureAwait(false);
         }
 
+        public async Task<HttpResponse> SendGetWithTimeoutAsync(
+            Uri endpoint,
+            IDictionary<string, string> headers,
+            ICoreLogger logger,
+            int timeoutInMs)
```

Comment:
Use CancellationToken here instead of timeout.

Reply:
Updated